src/highlighters.cc

```diff
@@ -992,7 +992,7 @@ void show_whitespaces(HighlightContext context, DisplayBuffer& display_buffer, B
                                                   std::max(win_column - column, 0_col);
                         atom_it->replace(tab + String(tabpad[(CharCount)0], count - tab.column_length()));
                     }
-                    else if (cp == ' ')
+                    else if (cp == ' ' && (!only_trailing || it == end - 1))
```

Comment:
Small nitpicking: if I remember correctly, mawww uses English operators like `not`, `and`, `or`…